internal/build/phase_config_provider.go

```diff
@@ -108,11 +111,33 @@ func WithBinds(binds ...string) PhaseConfigProviderOperation {
 func WithDaemonAccess() PhaseConfigProviderOperation {
 	return func(provider *PhaseConfigProvider) {
 		WithRoot()(provider)
+
 		bind := "/var/run/docker.sock:/var/run/docker.sock"
+
+		if dockerHost, ok := os.LookupEnv("DOCKER_HOST"); ok {
+			if strings.HasPrefix(dockerHost, "unix://") {
+				bind = fmt.Sprintf("%s:/var/run/docker.sock", strings.TrimPrefix(dockerHost, "unix://"))
+			}
+			if strings.HasPrefix(dockerHost, "tcp://") {
```

Comment:
If we're adding in support for DOCKER_HOST with `tcp`, should we also add support in for `DOCKER_CERT_PATH`? 

https://github.com/moby/moby/blob/3ea762b9f6ba256cf51bd2c35988f0c48bccf0b0/client/client.go#L40

Reply:
It could be useful sometimes, but I think that most of the time user will run the `docker API`  on localhost with just `HTTP` not `HTTPS`. If user runs `docker API` accessible from the outside internet then the user should ensure it uses valid certificate  signed by global authority like `Let's Encrypt`.

Reply:
Good catch! This change makes this no longer work: https://medium.com/buildpacks/pack-with-a-remote-docker-daemon-41aab804b839

I regularly use `DOCKER_HOST` + `DOCKER_CERT_PATH` to target remote Windows (and `arm`) daemons with `pack` and that does get lost with this implementation. I'm really not sure there's a secure way to expose the client's mTLS certs and keys to a container, even it's it's through a containers trusted store.

This is leading me to preferring an explicit `pack build` option to enable an alternative `DOCKER_HOST` for a run container instead of inferring from the `pack`'s `DOCKER_HOST`.